src/core/qgsmaplayer.h

```diff
@@ -528,6 +528,96 @@ class CORE_EXPORT QgsMapLayer : public QObject
+     * we did not manage to save the default metadata.
+     * \returns a QString with any status messages
+     * \since QGIS 3.0
+     */
+    virtual QString saveDefaultMetadata( bool &resultFlag SIP_OUT );
+
+    /**
+     * Save the properties of this layer as a named metadata
+     * (either as a .qmd file on disk or as a
+     * record in the users style table in their personal qgis.db)
+     * \param uri the file name or other URI for the
+     * metadata file. First an attempt will be made to see if this
+     * is a file and save to that, if that fails the qgis.db metadata
+     * table will be used to create a metadata entry who's
+     * key matches the URI.
+     * \param resultFlag a reference to a flag that will be set to false if
+     * we did not manage to save the default metadata.
+     * \returns a QString with any status messages
+     * \since QGIS 3.0
+     */
+    QString saveNamedMetadata( const QString &uri, bool &resultFlag );
+
+    /**
+     * Retrieve a named metadata for this layer if one
+     * exists (either as a .qmd file on disk or as a
+     * record in the users style table in their personal qgis.db)
+     * \param uri - the file name or other URI for the
+     * metadata file. First an attempt will be made to see if this
+     * is a file and load that, if that fails the qgis.db metadata
+     * table will be consulted to see if there is a metadata who's
+     * key matches the URI.
+     * \param resultFlag a reference to a flag that will be set to false if
+     * we did not manage to load the default metadata.
+     * \returns a QString with any status messages
+     * \since QGIS 3.0
+     */
+    virtual QString loadNamedMetadata( const QString &uri, bool &resultFlag SIP_OUT );
+
+    /**
+     * Retrieve the default metadata for this layer if one
+     * exists (either as a .qmd file on disk or as a
+     * record in the users metadata table in their personal qgis.db)
+     * \param resultFlag a reference to a flag that will be set to false if
+     * we did not manage to load the default metadata.
+     * \returns a QString with any status messages
+     * \since QGIS 3.0
+     */
+    QString loadDefaultMetadata( bool &resultFlag );
+
+    /**
+     * Retrieve a named metadata for this layer from a sqlite database.
+     * \param db path to sqlite database
+     * \param uri uri for table
+     * \param qmd will be set to QMD xml metadata content from database
+     * \returns true if style was successfully loaded
+     * \since QGIS 3.0
+     */
+    bool loadNamedMetadataFromDatabase( const QString &db, const QString &uri, QString &qmd );
+
+    /**
+     * Import the metadata of this layer from a QDomDocument
+     * \param document source QDomDocument
+     * \param errorMessage this QString will be initialized on error
+     * \returns true on success
+     * \since QGIS 3.0
+     */
+    bool importNamedMetadata( QDomDocument &document, QString &errorMessage );
```

Comment:
To me "import" implies that current settings aren't lost, but just added to. Should this be renamed to "load" instead?

Reply:
I was doing exactly the same API as the one for style: https://qgis.org/api/classQgsMapLayer.html#a59f50a11ff98460ac9e6c8993bff6295
I agree with you, but I like having the same behavior/naming conventions between style and metadata.